language/stdlib/modules/LibraAccount.move

```diff
@@ -569,73 +611,118 @@ module LibraAccount {
     /// `auth_key_prefix` | `new_account_address`, for non synthetic CoinType.
     /// Creates Preburn resource under account 'new_account_address'
     public fun create_designated_dealer<CoinType>(
-        association: &signer,
+        treasury_compliance_role: &RoleId,
+        dd_creation_capability: &Capability<CreateDesignatedDealer>,
+        parent_vasp_creation_capability: &Capability<CreateParentVASP>,
+        preburn_creation_capability: &Capability<CreatePreburn>,
         new_account_address: address,
         auth_key_prefix: vector<u8>,
-    ) {
-        // TODO: this should check for AssocRoot in the future
-        Association::assert_is_association(association);
+    ) acquires AccountOperationsCapability {
         let new_dd_account = create_signer(new_account_address);
         Event::publish_generator(&new_dd_account);
-        Libra::publish_preburn_to_account<CoinType>(association, &new_dd_account);
-        DesignatedDealer::publish_designated_dealer_credential(association, &new_dd_account);
-        let role_id = 2;
-        make_account<CoinType>(new_dd_account, auth_key_prefix, false, role_id)
+        Libra::publish_preburn_to_account<CoinType>(preburn_creation_capability, &new_dd_account);
+        DesignatedDealer::publish_designated_dealer_credential(dd_creation_capability, &new_dd_account);
+        let role_creation_cap = &borrow_global<AccountOperationsCapability>(
+            CoreAddresses::ASSOCIATION_ROOT_ADDRESS()
+        ).role_creation_cap;
+        let role = Roles::new_role(
+            treasury_compliance_role,
+            role_creation_cap,
+            &new_dd_account,
+            Roles::DESIGNATED_DEALER_ROLE_ID(),
+        );
+        Roles::grant_parent_vasp_privileges(
```

Comment:
Intended?

Reply:
yes, but I was just following the code that was already there, so happy to take it out :)